tests/test_samples.py

```diff
@@ -638,3 +638,38 @@ def test_posterior_points():
     ns = NestedSamples(root='./tests/example_data/pc')
     assert_array_equal(ns.posterior_points(), ns.posterior_points())
     assert_array_equal(ns.posterior_points(0.5), ns.posterior_points(0.5))
+
+
+def test_logzero_mask_prior_level():
+    np.random.seed(3)
+    ns0 = NestedSamples(root='./tests/example_data/pc')
+    pi0 = ns0.set_beta(0)
+    NS0 = ns0.ns_output(nsamples=2000)
+    mask = ((ns0.x0 > -0.3) & (ns0.x2 > 0.2) & (ns0.x4 < 3.5)).to_numpy()
+
+    V_prior = pi0[mask].weights.sum() / pi0.weights.sum()
+    V_posterior = ns0[mask].weights.sum() / ns0.weights.sum()
+    logZ_V_mask = NS0.logZ.mean() + np.log(V_posterior) - np.log(V_prior)
+
+    ns1 = merge_nested_samples((ns0[mask],))
+    NS1 = ns1.ns_output(nsamples=2000)
+
+    assert abs(NS1.logZ.mean() - logZ_V_mask) < NS1.logZ.std()
+
+
+def test_logzero_mask_likelihood_level():
+    np.random.seed(3)
+    ns0 = NestedSamples(root='./tests/example_data/pc')
+    NS0 = ns0.ns_output(nsamples=2000)
+    mask = ((ns0.x0 > -0.3) & (ns0.x2 > 0.2) & (ns0.x4 < 3.5)).to_numpy()
+
+    V_posterior = ns0[mask].weights.sum() / ns0.weights.sum()
+    logZ_V_mask = NS0.logZ.mean() + np.log(V_posterior)
+
+    ns1 = merge_nested_samples((ns0,))
+    ns1.logL = np.where(mask, ns1.logL, -np.inf)
+    # ns1.logL_birth = np.where(mask, ns1.logL_birth, -np.inf)
+    ns1 = merge_nested_samples((ns1,))
+    NS1 = ns1.ns_output(nsamples=2000)
+
+    assert abs(NS1.logZ.mean() - logZ_V_mask) < NS1.logZ.std()
```

Comment:
This test assigns `-inf` to the `logL` outside the mask. 

In its current form this doesn't even compute the new sample. It throws an off-by-one error on line 672:
```python
ValueError: Length of values (1501) does not match length of index (1500)
```
This off-by-one error is interesting in the light of #119, however, #119 does not seem to fix this, it just changes the lengths to 1500 and 1499.

If I uncomment line 671, then it can actually compute the new samples. However the resulting evidence is quite off from the posterior volume re-weighted one. Should I be setting both `logL` _and_ `logL_birth` to `-inf`?

Reply:
This corresponds to the orange line in the MWE plots.